circuit-lib/circuit-lib.circom/scripts/buildTestCircuit.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/sh
+set -e
+
+# --------------------------------------------------------------------------------
+# Phase 2
+# ... circuit-specific stuff
+
+# Check if artifacts exists
+if [ -d "./artifacts/$1" ]; then
+  # If it exists, delete it and its contents
+  rm -r "./artifacts/$1"
+fi
+
+# Create a directory for the circuit
+mkdir -p "./artifacts/$1"
+
+
+# Compile circuit => $1 will refer to the full circuit name
+CIRCUIT_NAME="${1%%_*}"
+circom ./tests/**/**/$1.circom -o ./artifacts/$1 --r1cs --wasm -l node_modules/circomlib/circuits
+
+#Setup
+pnpm snarkjs groth16 setup ./artifacts/$1/$1.r1cs ./artifacts/ptau/pot19.ptau ./artifacts/$1/$CIRCUIT_NAME.zkey
+
+# # Generate reference zkey
+pnpm snarkjs zkey new ./artifacts/$1/$1.r1cs ./artifacts/ptau/pot19.ptau ./artifacts/$1/$1_0000.zkey
+
+# # Ceremony just like before but for zkey this time
+pnpm snarkjs zkey contribute ./artifacts/$1/$1_0000.zkey ./artifacts/$1/$1_0001.zkey \
+    --name="First $1 contribution" -v -e="$(head -n 4096 /dev/urandom | openssl sha1)"
+
+pnpm snarkjs zkey contribute ./artifacts/$1/$1_0001.zkey ./artifacts/$1/$1_0002.zkey \
+    --name="Second $1 contribution" -v -e="$(head -n 4096 /dev/urandom | openssl sha1)"
+
+pnpm snarkjs zkey contribute ./artifacts/$1/$1_0002.zkey ./artifacts/$1/$1_0003.zkey \
+    --name="Third $1 contribution" -v -e="$(head -n 4096 /dev/urandom | openssl sha1)"
```

Comment:
There is no need to do more than one contribution for testing. Please remove the additional contributions to reduce compile time.